cachito/workers/pkg_managers/general.py

```diff
@@ -179,18 +179,18 @@ async def async_download_binary_file(
     :raise NetworkError: If download failed
     """
     try:
-        log.debug(f"Download started - {tarball_name}")
         async with session.get(url, auth=auth, raise_for_status=True) as resp:
             with open(os.path.join(download_dir, tarball_name), "wb") as f:
                 while True:
                     chunk = await resp.content.read(chunk_size)
                     if not chunk:
                         break
                     f.write(chunk)
-    except Exception as e:
-        raise NetworkError(f"Could not download {url}: {e}")
 
-    log.debug(f"Download completed - {tarball_name}")
+    except Exception as e:
+        log.error(f"Unsuccessful download: {tarball_name}")
+        message = f"Could not download. exception_name: {e.__class__.__name__}, details: {str(e)}"
+        raise NetworkError(message) from None
```

Comment:
Can we comment here reason why we want to suppress exc context?

Reply:
What do you mean by suppressing exc context?

Reply:
I believe Martin is asking about the `from None` part.

Reply:
I'm wondering if we're doing this backwards. Maybe the error message doesn't need to be that verbose (keep just `f"Could not download {tarball_name} from {url}"`) and we can _not_ suppress the context instead?

On the other hand, having everything in the same message might make it easier to search in Splunk logs etc.

Reply:
The problem, raising our exception inside the other, makes us receive in the logs that Python warning:    

    During handling of the above exception, another exception occurred 

That way, cachito logs are much more comfortable to read using `from None`.

Reply:
The alternative is `raise ... from exception`, which is intended exactly for the use case of transforming errors into our custom error types
https://docs.python.org/3/tutorial/errors.html#exception-chaining

Although admittedly, it's not a whole lot better, AFAIK it just replaces
```
During handling of the above exception, another exception occurred:
```
with
```
The above exception was the direct cause of the following exception:
```